apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -207,6 +215,12 @@ class SectionRow extends Component {
     } = this.props;
     const { editing, deleting } = this.state;
 
+    // When deleting a section, I've occasionally seen us attempt a render with
+    // a null section for some reason. This is here to provide defense against this.
+    if (!section) {
+      return null;
+    }
```

Comment:
Hm, I see what you mean.  I think I agree with (b) in that we should be leaning on React to do its job and render fast, and revisit `componentShouldUpdate`-type optimizations later when we notice performance problems. Clean up in future PR is great :+1: